tests/storage/snapshots/conftest.py

```
@@ -149,3 +153,199 @@ def vm_restore_with_predictable_names(
+            vms.append(vm)
+        yield vms
+    finally:
+        cleanup_errors = []
+        for vm in vms:
+            try:
+                vm.clean_up()
+            except Exception as exp:
+                LOGGER.warning(f"Failed to clean up VM {vm.name}: {exp}")
+                cleanup_errors.append(exp)
+        if cleanup_errors:
+            raise ExceptionGroup("VM cleanup failures", cleanup_errors)
```

Ruff (0.16.2)

[warning] Avoid specifying long messages outside the exception class

(TRY003)